<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1350</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Image Viewer</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QMdiArea" name="mdiArea"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1350</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionRevert"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuImage">
    <property name="title">
     <string>Image</string>
    </property>
    <addaction name="actionImage_Info"/>
    <addaction name="separator"/>
    <addaction name="actionInvert"/>
    <addaction name="actionSharpen"/>
    <addaction name="actionSmooth"/>
    <addaction name="actionColor_Adjust"/>
    <addaction name="separator"/>
    <addaction name="actionCrop"/>
    <addaction name="actionResize"/>
    <addaction name="separator"/>
    <addaction name="actionFit_window"/>
    <addaction name="actionZoom"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuImage"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="windowTitle">
    <string>Image operations</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen"/>
   <addaction name="separator"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
   <addaction name="actionRevert"/>
   <addaction name="separator"/>
   <addaction name="actionImage_Info"/>
   <addaction name="separator"/>
   <addaction name="actionInvert"/>
   <addaction name="separator"/>
   <addaction name="actionSharpen"/>
   <addaction name="separator"/>
   <addaction name="actionEdge_detect"/>
   <addaction name="separator"/>
   <addaction name="actionSmooth"/>
   <addaction name="separator"/>
   <addaction name="actionColor_Adjust"/>
   <addaction name="separator"/>
   <addaction name="actionCrop"/>
   <addaction name="separator"/>
   <addaction name="actionResize"/>
   <addaction name="separator"/>
   <addaction name="actionUndo"/>
   <addaction name="separator"/>
   <addaction name="actionRedo"/>
   <addaction name="separator"/>
   <addaction name="actionRotate"/>
   <addaction name="separator"/>
  </widget>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="mysource.qrc">
     <normaloff>:/images/open2f.PNG</normaloff>:/images/open2f.PNG</iconset>
   </property>
   <property name="text">
    <string>Open...</string>
   </property>
   <property name="statusTip">
    <string>Open an image from a file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionInvert">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="source.qrc">
     <normaloff>:/new/prefix1/nagate.PNG</normaloff>:/new/prefix1/nagate.PNG</iconset>
   </property>
   <property name="text">
    <string>Negate</string>
   </property>
   <property name="statusTip">
    <string>Negate the image color</string>
   </property>
  </action>
  <action name="actionSharpen">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="sour.qrc">
     <normaloff>:/new/prefix1/sharpen.PNG</normaloff>:/new/prefix1/sharpen.PNG</iconset>
   </property>
   <property name="text">
    <string>Sharpen</string>
   </property>
   <property name="statusTip">
    <string>Sharpen the image</string>
   </property>
  </action>
  <action name="actionSmooth">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="mysource.qrc">
     <normaloff>:/images/gaussf.PNG</normaloff>:/images/gaussf.PNG</iconset>
   </property>
   <property name="text">
    <string>Smooth</string>
   </property>
   <property name="statusTip">
    <string>Smooth the image</string>
   </property>
  </action>
  <action name="actionZoom">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="mysource.qrc">
     <normaloff>:/images/zoom1f.PNG</normaloff>:/images/zoom1f.PNG</iconset>
   </property>
   <property name="text">
    <string>Zoom...</string>
   </property>
   <property name="statusTip">
    <string>Change the viewing zoom</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Z</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset resource="source.qrc">
     <normaloff>:/images/reportd.PNG</normaloff>:/images/reportd.PNG</iconset>
   </property>
   <property name="text">
    <string>About</string>
   </property>
   <property name="statusTip">
    <string>About this program</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
   <property name="menuRole">
    <enum>QAction::AboutRole</enum>
   </property>
  </action>
  <action name="actionQuit">
   <property name="icon">
    <iconset resource="mysource.qrc">
     <normaloff>:/images/exit.PNG</normaloff>:/images/exit.PNG</iconset>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="statusTip">
    <string>Exit the program</string>
   </property>
   <property name="menuRole">
    <enum>QAction::QuitRole</enum>
   </property>
  </action>
  <action name="actionSave">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="mysource.qrc">
     <normaloff>:/images/savef.PNG</normaloff>:/images/savef.PNG</iconset>
   </property>
   <property name="text">
    <string>Save As...</string>
   </property>
   <property name="statusTip">
    <string>Save the image</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionImage_Info">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="source.qrc">
     <normaloff>:/images/reportd.PNG</normaloff>:/images/reportd.PNG</iconset>
   </property>
   <property name="text">
    <string>Image Info</string>
   </property>
   <property name="statusTip">
    <string>Show information about the image</string>
   </property>
  </action>
  <action name="actionRevert">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="mysource.qrc">
     <normaloff>:/images/resetf.PNG</normaloff>:/images/resetf.PNG</iconset>
   </property>
   <property name="text">
    <string>Reset</string>
   </property>
   <property name="toolTip">
    <string>Reset</string>
   </property>
   <property name="statusTip">
    <string>Revert back to the file on disk</string>
   </property>
  </action>
  <action name="actionCrop">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="mysource.qrc">
     <normaloff>:/images/cropf.PNG</normaloff>:/images/cropf.PNG</iconset>
   </property>
   <property name="text">
    <string>Crop</string>
   </property>
   <property name="statusTip">
    <string>Crop the image based on the selection</string>
   </property>
  </action>
  <action name="actionColor_Adjust">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="source.qrc">
     <normaloff>:/images/color.PNG</normaloff>:/images/color.PNG</iconset>
   </property>
   <property name="text">
    <string>Color Adjust...</string>
   </property>
  </action>
  <action name="actionResize">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="source.qrc">
     <normaloff>:/images/resize2.PNG</normaloff>:/images/resize2.PNG</iconset>
   </property>
   <property name="text">
    <string>Resize...</string>
   </property>
   <property name="statusTip">
    <string>Resize the image</string>
   </property>
  </action>
  <action name="actionFit_window">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="source.qrc">
     <normaloff>:/images/fit2.PNG</normaloff>:/images/fit2.PNG</iconset>
   </property>
   <property name="text">
    <string>Fit window</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="mysource.qrc">
     <normaloff>:/images/undo10.PNG</normaloff>:/images/undo10.PNG</iconset>
   </property>
   <property name="text">
    <string>undo</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="mysource.qrc">
     <normaloff>:/images/redo10.PNG</normaloff>:/images/redo10.PNG</iconset>
   </property>
   <property name="text">
    <string>redo</string>
   </property>
  </action>
  <action name="actionRotate">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="mysource.qrc">
     <normaloff>:/images/rotatef.PNG</normaloff>:/images/rotatef.PNG</iconset>
   </property>
   <property name="text">
    <string>rotate</string>
   </property>
  </action>
  <action name="actionEdge_detect">
   <property name="icon">
    <iconset resource="source.qrc">
     <normaloff>:/new/prefix1/edge2f.PNG</normaloff>:/new/prefix1/edge2f.PNG</iconset>
   </property>
   <property name="text">
    <string>edge detect</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="mysource.qrc"/>
  <include location="source.qrc"/>
  <include location="sour.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>doOpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>doAbout()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>455</x>
     <y>12</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSharpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>doSharpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionInvert</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>doNegate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mdiArea</sender>
   <signal>subWindowActivated(QMdiSubWindow*)</signal>
   <receiver>MainWindow</receiver>
   <slot>doChangeImage(QMdiSubWindow*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>150</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>doSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionZoom</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>doZoom()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionImage_Info</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>doInfo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCrop</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>doCrop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRevert</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>doRevert()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSmooth</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>doSmooth()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionColor_Adjust</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>doSliders()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionResize</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>doResize()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>imageEditable(bool)</signal>
   <receiver>actionSave</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>imageEditable(bool)</signal>
   <receiver>actionRevert</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>imageEditable(bool)</signal>
   <receiver>actionImage_Info</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>imageEditable(bool)</signal>
   <receiver>actionInvert</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>imageEditable(bool)</signal>
   <receiver>actionSmooth</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>imageEditable(bool)</signal>
   <receiver>actionSharpen</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>imageEditable(bool)</signal>
   <receiver>actionCrop</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>imageEditable(bool)</signal>
   <receiver>actionResize</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>imageEditable(bool)</signal>
   <receiver>actionZoom</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionFit_window</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>doFillWindow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>270</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>imageEditable(bool)</signal>
   <receiver>actionFit_window</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>270</x>
     <y>200</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>imageEditable(bool)</signal>
   <receiver>actionColor_Adjust</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>343</x>
     <y>218</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>imageEditable(bool)</signal>
  <slot>doOpen()</slot>
  <slot>doNegate()</slot>
  <slot>doSharpen()</slot>
  <slot>doAbout()</slot>
  <slot>doChangeImage(QMdiSubWindow*)</slot>
  <slot>doSave()</slot>
  <slot>doZoom()</slot>
  <slot>doInfo()</slot>
  <slot>doCrop()</slot>
  <slot>doRevert()</slot>
  <slot>doSmooth()</slot>
  <slot>doSliders()</slot>
  <slot>doResize()</slot>
  <slot>doFillWindow()</slot>
 </slots>
</ui>
